EmotiBit.cpp

```diff
@@ -1120,18 +1120,85 @@ bool EmotiBit::addPacket(uint32_t timestamp, EmotiBit::DataType t, float * data,
 		//if (t == EmotiBit::DataType::DATA_OVERFLOW){
 		//	addDebugPacket((uint8_t)EmotiBit::DebugTags::WIFI_CONNHISTORY, timestamp);  // addDebugPacket(case, timestamp) 
 		//}
+		uint8_t protocolVersion = 1;
+		if (DC_DO_V2)
+		{
+			if (t == EmotiBit::DataType::DATA_CLIPPING || t == EmotiBit::DataType::DATA_OVERFLOW)
+			{
+				protocolVersion = 2;
+			}
+		}
 
-		header = EmotiBitPacket::createHeader(typeTags[(uint8_t)t], timestamp, _outDataPacketCounter++, dataLen);
+
+		header = EmotiBitPacket::createHeader(typeTags[(uint8_t)t], timestamp, _outDataPacketCounter++, dataLen, protocolVersion);
 		_outDataPackets += EmotiBitPacket::headerToString(header);
-		for (uint16_t i = 0; i < dataLen; i++) {
-			_outDataPackets += ",";
-			if (t == EmotiBit::DataType::DATA_CLIPPING || t == EmotiBit::DataType::DATA_OVERFLOW) {
-				// If it's a clipping/overflow type, write the data as a string rather than float
-				// ToDo: consider how to better keep track of clipping and overflows
-				_outDataPackets += typeTags[(uint8_t)data[i]];
+
+		if (t == EmotiBit::DataType::DATA_CLIPPING || t == EmotiBit::DataType::DATA_OVERFLOW) {
+			// Handle clippping and overflow as a special case
+			for (uint8_t i = 0; i < (uint8_t)EmotiBit::DataType::length; i++)
+			{
+				// Add all the clipping/overflow events across all the buffers to the packet
+				if (i != (uint8_t) EmotiBit::DataType::DATA_CLIPPING && i != (uint8_t) EmotiBit::DataType::DATA_OVERFLOW) {
+					// Skip clipping & overflow types
+					size_t count = 0;
+					if (t == EmotiBit::DataType::DATA_CLIPPING)
+					{
+						count = dataDoubleBuffers[i]->getClippedCount(DoubleBufferFloat::BufferSelector::OUT);
+					}
+					if (t == EmotiBit::DataType::DATA_OVERFLOW)
+					{
+						count = dataDoubleBuffers[i]->getOverflowCount(DoubleBufferFloat::BufferSelector::OUT);
+					}
+					if (DC_DO_V2)
```

Comment:
Is there a reason it is conditioned on this `const` bool set as true in class?
If we are moving forward with this new protocol version, should we not just remove the conditional and use only the new protocol?